### PR TITLE
[Fixed] Dotnet version for DevContainer

### PR DESCRIPTION
**Pull request by cschuchardt88**
# Description

The version of `dotnet` was wrong for the `devcontainer`.

image

## Type of change

- [ ] Optimization (the change is only an optimization)
- [ ] Style (the change is only a code style for better maintenance or standard purpose)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
- Locally

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by vncoelho — COMMENTED** [review]
Good!
now global does not need to be modified locally.

Can you add the link to the list of images?
Then we can follow easily next update.

**Comment by cschuchardt88** [comment]
images and what link?

File: .devcontainer/devcontainer.dockerfile [hunk 1]
```diff
@@ -1,3 +1,4 @@
-FROM mcr.microsoft.com/devcontainers/dotnet:9.0-noble
+FROM mcr.microsoft.com/dotnet/sdk:9.0.101-noble
```
Comment (vncoelho) [inline .devcontainer/devcontainer.dockerfile:1]: Add a comment with:

`# https://github.com/dotnet/dotnet-docker/blob/main/README.sdk.md`

**Review by shargon — APPROVED** [review]

**Review by nan01ab — APPROVED** [review]

**Comment by Jim8y** [comment]
lets revert back to dotnet 8 please, cant use dotnet 9 for now.

**Comment by cschuchardt88** [comment]
That is unnecessary update coming out next month. Unless we have a new release coming out this month. then no reason to revert it.

**Review by AnnaShaleva — APPROVED** [review]

**Review by AnnaShaleva — COMMENTED** [review]
> You can also fix the issue, if you compile the neo.vm to dotnet standard then import it.

But are you sure that the neo core itself is not affected by this bug in dotnet 9? Because BigInteger arithmetic is widely used in the core (not only in VM).

**Comment by cschuchardt88** [comment]
> But are you sure that the neo core itself is not affected by this bug in dotnet 9? Because BigInteger arithmetic is widely used in the core (not only in VM).

Only thing that isnt working is `GetBitLength` or invalid bit length, and in the `neo.vm` is the really the only the place this happens.

**Review by AnnaShaleva — COMMENTED** [review]
> and in the neo.vm is the really the only the place this happens.

If it's true, then I'm not against of this PR.

But in general I'd expect the same dotnet version to be used within the whole stack. It's the way how we build software in `nspcc-dev` org, and it's a good practice, the whole stack has the same minimum required Go version, so I'd expect the same from the `neo-project` org. So if we take a decision to revert to dotnet 8, then every project should be reverted.

**Comment by cschuchardt88** [comment]
> > and in the neo.vm is the really the only the place this happens.
> 
> If it's true, then I'm not against of this PR.
> 
> But in general I'd expect the same dotnet version to be used within the whole stack. It's the way how we build software in `nspcc-dev` org, and it's a good practice, the whole stack has the same minimum required Go version, so I'd expect the same from the `neo-project` org. So if we take a decision to revert to dotnet 8, then every project should be reverted.

Looks like elliptical curve uses `GetBitLength` too. However all test are showing the correct output. So i dont know what `neo-devpack` is doing to have trouble with `BigInteger`.

**Review by Jim8y — APPROVED** [review]

**Pull request merged**